hummingbot/core/cpp/OrderBookEntry.cpp

```diff
@@ -40,16 +40,16 @@ void truncateOverlapEntries(std::set<OrderBookEntry> &bidBook, std::set<OrderBoo
 void truncateOverlapEntriesDex(std::set<OrderBookEntry> &bidBook, std::set<OrderBookEntry> &askBook) {
     std::set<OrderBookEntry>::reverse_iterator bidIterator = bidBook.rbegin();
     std::set<OrderBookEntry>::iterator askIterator = askBook.begin();
-    while (bidIterator != bidBook.rend() && askIterator != askBook.end()) {
+    while (bidIterator != bidBook.rend() && askIterator != askBook.end() && !bidBook.empty() && !askBook.empty()) {
         const OrderBookEntry& topBid = *bidIterator;
         const OrderBookEntry& topAsk = *askIterator;
         if (topBid.price >= topAsk.price) {
             if (topBid.amount*topBid.price > topAsk.amount*topAsk.price) {
                 askBook.erase(askIterator++);
             } else {
                 std::set<OrderBookEntry>::iterator eraseIterator = (std::next(bidIterator)).base();
-                bidIterator++;
                 bidBook.erase(eraseIterator);
+                bidIterator++;
```

Comment:
does this same change need to be done to `truncateOverlapEntriesCentralised`?

Reply:
It does not affect, but I changed it to maintain the format across.